dlib/cuda/tensor_tools.h

```diff
@@ -20,6 +20,9 @@ namespace dlib
     bool dnn_prefer_fastest_algorithms();
     void set_dnn_prefer_fastest_algorithms();
     void set_dnn_prefer_smallest_algorithms();
+
+    bool use_cuda();
+    void set_use_cuda(bool flag);
```

Comment:
Add any new public functions to the docs (the _abstract.h files).  These should be documented next to `set_dnn_prefer_smallest_algorithms()`, so over in `dlib/dnn/core_abstract.h`.